File: copl/com.elster.jupiter.osgi.goodies/src/main/java/com/elster/jupiter/osgi/goodies/Association.java
/*
 * Copyright (c) 2017 by Honeywell International Inc. All Rights Reserved
 */

package com.elster.jupiter.osgi.goodies;

import java.util.Map;

import com.elster.jupiter.orm.associations.RefAny;

public class Association {
	private final String name;
	private final Node from;
	private final Node to;
	
	Association(String name, Node from , Node to) {
		this.name = name;
		this.from = from;
		this.to = to;
	}

	public void append(StringBuilder builder,Map<RefAny,Node> inventory) {
		builder.append("\t");
		builder.append(Node.quote(from.toString()));
		builder.append(" -> ");
		builder.append(Node.quote(to.toString()));
		builder.append(" [label=");
		builder.append(Node.quote(name));
		builder.append("]\n");
		to.append(builder,inventory);
	}
	
	Node getTo() {
		return to;
	}
	
}

File: copl/com.elster.jupiter.bpm.extjs/src/main/web/js/bpm/src/processes/model/Association.js
/*
 * Copyright (c) 2019 by Honeywell International Inc. All Rights Reserved
 */

Ext.define('Bpm.processes.model.Association', {
    extend: 'Ext.data.Model',
    requires: [
        'Uni.property.model.Property',
    ],
    fields: [
        {name: 'type', type: 'string'},
        {name: 'name', type: 'string'}
    ],
    associations: [
        {
            type: 'hasMany',
            name: 'properties',
            model: 'Uni.property.model.Property',
            associationKey: 'properties',
            foreignKey: 'properties',
            getTypeDiscriminator: function (node) {
                return 'Uni.property.model.Property';
            }
        },
    ]
});